machine/v1beta1/types_gcpprovider.go

```diff
@@ -54,9 +54,20 @@ type GCPMachineProviderSpec struct {
 	// ProjectID is the project in which the GCP machine provider will create the VM.
 	// +optional
 	ProjectID string `json:"projectID,omitempty"`
+	// GuestAccelerators is a list of GPUs to be attached to the VM.
+	// +optional
+	GuestAccelerators []*GCPAcceleratorConfig `json:"guestAccelerators,omitempty"`
```

Comment:
Does this need to be a slice of pointers?  Any reason not to use `[]GCPAcceleratorConfig`?

Reply:
follows https://pkg.go.dev/google.golang.org/api/compute/v1

Reply:
> follows https://pkg.go.dev/google.golang.org/api/compute/v1 convention

We expect our openshift APIs to follow openshift and kubernetes conventions for a consistent look and feel. Can you describe why its important to allow an element to nil? What additional power does it grant the user of this API?

Reply:
I don't think there's any reason we need this to be a point, please change this one Sam